timemachine/cpp/src/kernels/k_exchange.cu

```diff
@@ -59,76 +65,47 @@ void __global__ k_setup_proposals(
     }
 }
 
-template <typename RealType>
-void __global__ k_attempt_exchange_move(
-    const int N,
-    const RealType *__restrict__ rand,           // [1]
-    const RealType *__restrict__ before_max,     // [1]
-    const RealType *__restrict__ before_log_sum, // [1]
-    const RealType *__restrict__ after_max,      // [1]
-    const RealType *__restrict__ after_log_sum,  // [1]
-    const int *__restrict__ mol_offsets,         // [num_mols + 1]
-    const int *__restrict__ selected_mol,        // [1]
-    const double *__restrict__ moved_coords,     // [num_atoms_in_each_mol, 3]
-    double *__restrict__ dest_coords,            // [N, 3]
-    size_t *__restrict__ num_accepted            // [1]
+void __global__ k_accepted_exchange_move(
+    const int batch_size,
+    const int num_atoms_in_each_mol,
+    const int *__restrict__ accepted_batched_move, // [1]
+    const int *__restrict__ mol_idx_per_batch,     // [batch_size]
+    const int *__restrict__ mol_offsets,           // [num_target_mols]
+    const double *__restrict__ moved_coords,       // [batch_size, num_atoms_in_each_mol, 3]
+    double *__restrict__ dest_coords,              // [N, 3]
+    size_t *__restrict__ num_accepted,             // [1],
+    int *__restrict__ rand_offset                  // [1]
 ) {
     // Note that this kernel does not handle multiple proposals, expects that the proposals
     // have been reduced down to a single proposal beforehand.
-    int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    const int idx = blockIdx.x * blockDim.x + threadIdx.x;
     assert(idx == 0);
 
-    const int moved_mol_idx = selected_mol[0];
-    const int mol_start = mol_offsets[moved_mol_idx];
-    const int mol_end = mol_offsets[moved_mol_idx + 1];
-    const int num_atoms_in_each_mol = mol_end - mol_start;
-
-    const RealType before_log_prob =
-        convert_nan_to_inf<RealType>(compute_logsumexp_final<RealType>(before_max[0], before_log_sum[0]));
-    const RealType after_log_prob =
-        convert_nan_to_inf<RealType>(compute_logsumexp_final<RealType>(after_max[0], after_log_sum[0]));
-
-    const RealType log_acceptance_prob = min(before_log_prob - after_log_prob, static_cast<RealType>(0.0));
-    const bool accepted = rand[0] < exp(log_acceptance_prob);
-    if (idx == 0 && accepted) {
+    const int batch_idx = accepted_batched_move[0];
+    // If the selected batch idx is not less then the total batch size, no proposal was accepted, we can exit immediately.
```

Comment:
then -> than

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/1274/commits/b80d8d181a8da212087f4c07778dbf56a9f241d4